aeron-client/src/main/java/io/aeron/logbuffer/LogBufferDescriptor.java

```diff
@@ -878,4 +899,29 @@ public static int computeAssembledFrameLength(final int length, final int maxPay
 
         return HEADER_LENGTH + (numMaxPayloads * maxPayloadSize) + remainingPayload;
     }
+
+    /**
+     * Set the channel URI in the metadata buffer as a length-prefixed ASCII-string.
+     *
+     * @param metadataBuffer containing the meta data.
+     * @param channelUri      the channel URI to set.
+     * @throws IllegalArgumentException if the URI exceeds the maximum length.
+     */
+    public static void channelUri(final UnsafeBuffer metadataBuffer, final String channelUri) {
+        if (channelUri.length() > LOG_CHANNEL_URI_MAX_LENGTH) {
```

Comment:
Should use `ChannelUri.MAX_URI_LENGTH` instead.

Reply:
Fixed

Reply:
I introduced a new constant:

```
    LOG_CHANNEL_URI_MAX_LENGTH = LOG_META_DATA_LENGTH - (LOG_CHANNEL_URI_OFFSET + SIZE_OF_INT);
```

This is now used to validate that the stored channel-uri isn't too long. This value is now 7164. Perhaps I should remove 1 more to deal with /0 C strings.

The extra length is needed because we'll be adding settings to the user specified channel-uri, so we could exceed the current 4095 limit.